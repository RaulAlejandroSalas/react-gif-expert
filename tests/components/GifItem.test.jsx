const { render, screen } = require("@testing-library/react")
const { GitItem } = require("../../src/components/GitItem")

describe('Test in <GifItem.jsx/>', () => {
    const expectedTitle = 'Naruto'
    const expectedUrl   = 'https://naruto.com/naturo1.jpg'
    
    test('should match with the Snapshot', () => { 
        const {container} = render(<GitItem title={expectedTitle} url={expectedUrl}/>)
        expect(container).toMatchSnapshot()
     })

    test('should show the image with URL and ALT properties', () => {
        render(<GitItem title={expectedTitle} url={expectedUrl}/>)
        //screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(expectedUrl)
        expect(alt).toBe(expectedTitle)
     })

    test('should show the Component Text', () =>{
        render(<GitItem title={expectedTitle} url={expectedUrl}/>)
        expect(screen.getByText(expectedTitle)).toBeTruthy()
    })

 })