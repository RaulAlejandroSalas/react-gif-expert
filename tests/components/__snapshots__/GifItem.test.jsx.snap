// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem.jsx/> should match with the Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com/naturo1.jpg"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
